packages/compass-schema/src/modules/schema-analysis.spec.ts

```diff
@@ -1,18 +1,30 @@
 import sinon from 'sinon';
 import bson from 'bson';
 import { expect } from 'chai';
+import { DataServiceImpl } from 'mongodb-data-service';
```

Comment:
Do we need this dependency here? Seems like in tests all the relevant methods are mocked

Reply:
I don't think we need it, I like to use the dependency in the tests because it makes them fail sooner when the dependencies' types interfaces change. Even though the relevant methods are mocked, by using the type here we ensure our mock interfaces are returning the right types.
The typing in the non-test code should error when an interface changes, but sometimes overtime things might accidentally set to any and then more typing in our tests protects us against assuming an old interface.

Do you prefer the tests without it? Happy to remove it and pass something as `any` or `unknown as DataService`, no strong preference.

Reply:
Makes sense, updated to use `Partial<DataService> as unknown as DataService` so we get that double type safety and avoid the implementation usage.

Reply:
It’s definite okay with me :+1: `analyzeSchema` could also just take a `Pick<DataService, 'sample'>`, that would make this probably go away entirely and be a bit more precise as well (and would e.g. mean that if we start requiring another `DataService` method in `analyzeSchema`, we’d get type errors that remind us to update our tests etc.)